jujugui/static/gui/src/app/store/env/controller-api.js

```diff
@@ -973,6 +973,104 @@ YUI.add('juju-controller-api', function(Y) {
         request: 'Credential',
         params: {entities: entities}
       }, handler);
+    },
+
+    /**
+      Create or update a single cloud credential.
+
+      @method updateCloudCredential
+      @param {String} tag The Juju tag of the credential as a string, for
+        instance "cloudcred-google_dalek@local_google". Tags for credentials
+        are usually retrieved by calling the getTagsForCloudCredentials method.
+      @params {String} authType The authentication type, like "userpass",
+        "oauth2" or just "empty".
+      @params {Object} attrs Attributes containing credential values, as an
+        object mapping strings to strings.
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive an error message or null if the
+        credential creation/update succeeded.
+    */
+    updateCloudCredential: function(tag, authType, attrs, callback) {
+      // Decorate the user supplied callback.
+      const handler = function(userCallback, data) {
+        if (!userCallback) {
+          console.log(
+            'data returned by Cloud.UpdateCredentials API call:', data);
+          return;
+        }
+        if (data.error) {
+          userCallback(data.error);
+          return;
+        }
+        const results = data.response.results;
+        if (!results || results.length !== 1) {
+          // This should never happen.
+          userCallback(
+            'invalid results returned by Juju: ' + JSON.stringify(results));
+          return;
+        }
+        const err = results[0].error && results[0].error.message;
+        if (err) {
+          userCallback(err);
+          return;
+        }
+        userCallback(null);
```

Comment:
As described in the docstring, the callback only receives an error in this case. There is no valuable data in results (i.e. it is empty if everything went well, and we already parse possible errors).